allennlp/models/semantic_parsing/nlvr/nlvr_decoder_state.py

```diff
@@ -116,17 +89,11 @@ def combine_states(cls, states) -> 'NlvrDecoderState':
         scores = [score for state in states for score in state.score]
         rnn_states = [rnn_state for state in states for rnn_state in state.rnn_state]
         grammar_states = [grammar_state for state in states for grammar_state in state.grammar_state]
-        if states[0].terminal_actions is not None:
-            terminal_actions = [actions for state in states for actions in state.terminal_actions]
-            checklist_target = [target_list for state in states for target_list in
-                                state.checklist_target]
-            checklist_masks = [mask for state in states for mask in state.checklist_mask]
-            checklist = [checklist_list for state in states for checklist_list in state.checklist]
+        if states[0].checklist_state is not None:
```

Comment:
Does it make sense to just have the checklist state be `[None, None, ...]`, it there's no checklist, so you don't have to have this special case?

Reply:
Done